### ISSUE TITLE
TLS 1.3 vs future TLS versions

### ISSUE THREAD

**Issue by ms-s** [emu-wg/draft-ietf-emu-eap-tls13#26]
Section 2.1.5

In some sense, this figure depicting the no-peer-authentication case is
just an example specific to TLS 1.3 and is not necessarily
representative of the generic EAP-TLS case for potential future versions
of TLS.  Should we make that more clear from the prose?

    In the case where the EAP-TLS server rejects the ClientHello with a
    fatal error, the conversation will appear as shown in Figure 4.  The

Should we perhaps say that these examples are with TLS 1.3, even if the
actual normative protocol is no longer tied to a specific TLS version?

    The two paragraphs below replaces the corresponding paragraphs in
    Section 2.1.3 of [RFC5216] when EAP-TLS is used with TLS 1.3 or
    higher.  The other paragraphs in Section 2.1.3 of [RFC5216] still
    apply with the exception that SessionID is deprecated.

       If the EAP-TLS peer authenticates successfully, the EAP-TLS server
       MUST send an EAP-Request packet with EAP-Type=EAP-TLS containing
       TLS records conforming to the version of TLS used.  The message
       flow ends with the EAP-TLS server sending an EAP-Success message.

       If the EAP-TLS server authenticates successfully, the EAP-TLS peer
       MUST send an EAP-Response message with EAP-Type=EAP-TLS containing
       TLS records conforming to the version of TLS used.

Just to walk through the various scenarios here; if the last message of
the TLS handshake for a given TLS version is sent by the TLS client, that
will go in an EAP-Response, so the server is then able to send
EAP-Success directly.  On the other hand, if the last TLS handshake
message is sent by the server, that will have to go in an EAP-TLS
EAP-Request, and so the peer will have to send an empty EAP-TLS response
in order for the server to be able to wend EAP-Success?  Do we need to
have any text here to handle that empty EAP-TLS Eap-Request case?

**Comment by emanjon** [comment]
"In some sense, this figure depicting the no-peer-authentication case is
just an example specific to TLS 1.3"

I don't see why the no-peer-authentication figure is more TLS 1.3 specific that the other figures? 

"specific to TLS 1.3 and is not necessarily representative of the generic EAP-TLS case for potential future versions
of TLS. Should we make that more clear from the prose?"

I think that could be stated in general applying to basically the whole document, or?

**Comment by emanjon** [comment]
Seems like there is need for more text on what applies to TLS 1.2, TLS 1.3, and potential future versions.

"even if the actual normative protocol is no longer tied to a specific TLS version?"
I think the protocol was never tied to a specific version. RFC 5216 works with TLS 1.0, 1.1 and 1.2. I think the intention was that it would hopefully work with future versions as well, but TLS 1.3 is such a huge modification of TLS so quite many things in RFC 5216 does not apply any more.

**Comment by kaduk** [comment]
> Seems like there is need for more text on what applies to TLS 1.2, TLS 1.3, and potential future versions.

Okay.

(I think the latter half of the original comment/question is still open, though.)

**Comment by emanjon** [comment]
I added text that the message flows are TLS 1.3 specific.

"If the last message of the TLS handshake for a given TLS version is sent by the TLS client, that
will go in an EAP-Response, so the server is then able to send EAP-Success directly"

My understanding based on the information from the EAP expects is that close_notify would always have to be sent. So in the cases where there is no NewSessionTicket, you end up with a dummy round-trip.

I added text stating that close_notify is always sent.

EAP is a reequest respnse protocol. The procedures that the peer need to send a empty message back is handled by the EAP RFC so I don't think any additional text is needed.

Doing this update I noted that the first paragraph in section 2.1.1 is quite generic and not specific to "mutual authentication" I add a new issue regarding that, but I think this change can wait until AUTH48. One option is to move the paragraph out of the subsection, but the interactions with RFC5216 makes it tricky to move. Another option is to rename "mutual authentication" back to "base case" which is what it is called in RFC5216. These editorial changes can wait until AUTH48.
